cdc/model/schema_storage.go

```diff
@@ -72,38 +75,44 @@ func WrapTableInfo(schemaID int64, schemaName string, version uint64, info *mode
 		uniqueColumns:    make(map[int64]struct{}),
 		RowColumnsOffset: make(map[int64]int, len(info.Columns)),
 		ColumnsFlag:      make(map[int64]ColumnFlagType, len(info.Columns)),
-		handleColID:      -1,
+		handleColID:      []int64{-1},
 		HandleIndexID:    HandleIndexTableIneligible,
 		rowColInfos:      make([]rowcodec.ColInfo, len(info.Columns)),
+		rowColFieldTps:   make(map[int64]*types.FieldType, len(info.Columns)),
 	}
 
 	rowColumnsCurrentOffset := 0
 
 	for i, col := range ti.Columns {
 		ti.columnsOffset[col.ID] = i
-		isPK := false
+		pkIsHandle := false
 		if IsColCDCVisible(col) {
 			ti.RowColumnsOffset[col.ID] = rowColumnsCurrentOffset
 			rowColumnsCurrentOffset++
-			isPK = (ti.PKIsHandle && mysql.HasPriKeyFlag(col.Flag)) || col.ID == model.ExtraHandleID
-			if isPK {
+			pkIsHandle = (ti.PKIsHandle && mysql.HasPriKeyFlag(col.Flag)) || col.ID == model.ExtraHandleID
+			if pkIsHandle {
 				// pk is handle
-				ti.handleColID = col.ID
+				ti.handleColID = []int64{col.ID}
 				ti.HandleIndexID = HandleIndexPKIsHandle
 				ti.uniqueColumns[col.ID] = struct{}{}
 				ti.IndexColumnsOffset = append(ti.IndexColumnsOffset, []int{ti.RowColumnsOffset[col.ID]})
+			} else if ti.IsCommonHandle {
+				ti.HandleIndexID = HandleIndexPKIsHandle
+				ti.handleColID = ti.handleColID[:0]
+				pkIdx := tables.FindPrimaryIndex(info)
+				for _, pkCol := range pkIdx.Columns {
+					id := info.Columns[pkCol.Offset].ID
+					ti.handleColID = append(ti.handleColID, id)
+				}
 			}
+
 		}
 		ti.rowColInfos[i] = rowcodec.ColInfo{
```

Comment:
Do we need to set `VirtualGenCol` value here?

Reply:
No. `VirtualGenCol` is only used in `ChunkDecoder`, but I think setting this field is harmless. Updated.